extensions/wikia/WallNotifications/WallNotifications.class.php

```diff
@@ -386,103 +385,113 @@ protected function createKeyForMailNotification( $watcher, WallNotificationEntit
+				'targetUser' => $watcherName,
+				'fromAddress' => $this->app->wg->PasswordSender,
+				'replyToAddress' => $this->app->wg->NoReplyAddress,
+				'fromName' => $notification->data->msg_author_username,
+				'messageLink' => $notification->data->url,
+				'boardName' => $notification->data->article_title_text,
+				'threadTitle' => $notification->data->thread_title,
+				'wallUserName' => $notification->data->wall_username,
+				'authorUserName' => $notification->data->msg_author_username,
+				'threadId' => $notification->data->parent_id
+			];
 
-				if ( !( $watcher->getBoolOption('unsubscribed') === true ) ) {
-					$this->sendEmail( $watcher, $data );
-				}
-			}
+			F::app()->sendRequest( $controller, 'handle', $params );
 		}
 
 		return true;
 	}
 
-	protected function sendEmail( User $watcher, array $data ) {
-		global $wgPasswordSender, $wgNoReplyAddress;
+	protected function canSendToWatcher( User $watcher, $entityKey ) {
+		if ( $watcher->getId() == 0 ) {
+			return false;
+		}
+
+		// Don't send an email to users that unsubscribed their email address
+		if ( $watcher->getBoolOption( 'unsubscribed' ) === true ) {
+			return false;
+		}
 
-		$from = new MailAddress( $wgPasswordSender, 'Wikia' );
-		$replyTo = new MailAddress ( $wgNoReplyAddress );
+		$mode = $watcher->getOption( 'enotifwallthread' );
+		if ( empty( $mode ) ) {
+			return false;
+		}
 
-		$keys = array_keys( $data );
-		$values =  array_values( $data );
+		if ( $mode == WALL_EMAIL_EVERY ) {
+			return true;
+		}
 
-		$subject = wfMessage( $data['$MSG_KEY_SUBJECT'] )->inContentLanguage()->text();
+		if ( $mode == WALL_EMAIL_SINCEVISITED ) {
+			return empty( $this->uniqueUsers[$entityKey][$watcher->getId()] );
+		}
 
-		$text = wfMessage( $data['$MSG_KEY_BODY'] )->inContentLanguage()->text();
+		return false;
+	}
 
-		$subject = str_replace( $keys, $values, $subject );
+	protected function getEmailExtensionController( WallNotificationEntity $notification, $watcherName ) {
+		if ( !empty( $notification->data->article_title_ns )
+			&& MWNamespace::getSubject( $notification->data->article_title_ns ) == NS_WIKIA_FORUM_BOARD
+		) {
+			if ($notification->isMain()) {
+				$controller = 'Email\Controller\Forum';
+			}
+			else {
+				$controller = 'Email\Controller\ReplyForum';
+			}
+		} else if ( !$notification->isMain() ) {
+			$controller = 'Email\Controller\ReplyWallMessage';
+		} else if ( $notification->data->wall_username == $watcherName ) {
+			$controller = 'Email\Controller\OwnWallMessage';
+		} else {
+			$controller = 'Email\Controller\FollowedWallMessage';
+		}
 
-		$keys[] = '$SUBJECT';
-		$values[] = $subject;
+		return $controller;
+	}
 
-		$data['$SUBJECT'] = $subject;
-		$html = $this->app->getView( 'WallExternal', 'mail', ['data' => $data] )->render();
-		$text = str_replace( $keys, $values, $text );
+	/**
+	 * Get abstract from whole entity text.
+	 * HTML tags are stripped out, expect <p> <br> which are used to make it easier to read.
```

Comment:
Should probably be "HTML tags are stripped out, *except* \<p\> \<br\> which are used to make it easier to read."